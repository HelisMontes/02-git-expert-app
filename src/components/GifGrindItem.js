import React from 'react'

export const GifGrindItem = ({ title, url }) => {
    return (
        <div className='card'>
            <img src={url} alt='Title'/>
            <p>{title}</p>
        </div>
    )
}
