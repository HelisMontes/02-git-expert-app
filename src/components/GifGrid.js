import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGrindItem } from './GifGrindItem'
import PropTypes from 'prop-types';

export const GifGrid = ({ categories }) => {
    const { loading, data } = useFetchGifs( categories );
    return (
        <>
            <h3 className='animate__animated animate__flipInX'>{categories}</h3>
            {loading && <p className='animate__animated animate__flash'>Loading</p>}
            <div className='card-grind'>
                {
                    data.map( img =>(
                        <GifGrindItem 
                            key = {img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.protoType = {
    categories: PropTypes.string.isRequired
}