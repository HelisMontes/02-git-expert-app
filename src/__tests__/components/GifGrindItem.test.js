import React from 'react';
import { shallow } from "enzyme"
import { GifGrindItem } from "../../components/GifGrindItem"

describe ('Pruebas en componente GifGrid', () =>{
    const title = 'One punch'
    const url = 'https://media4.giphy.com/media/arbHBoiUWUgmc/giphy.gif?cid=2c52d32dg24wnnhv4e6alxkisi6xoa5r3vije6mppdkx6jjf&rid=giphy.gif'
    const wrapper = shallow( <GifGrindItem title={ title } url={ url } /> )

    test('Debe mostar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p').text().trim();
        expect( p ).toBe( title );
    })

    test('debe de tener la imagen igual al url y alt de los pops', () => {
        const img = wrapper.find('img')
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })
    
    test('debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__fadeIn') ).toBe(true);
    })
})
