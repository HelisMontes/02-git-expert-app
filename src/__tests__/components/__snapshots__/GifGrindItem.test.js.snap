// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGrid Debe mostar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One punch"
    src="https://media4.giphy.com/media/arbHBoiUWUgmc/giphy.gif?cid=2c52d32dg24wnnhv4e6alxkisi6xoa5r3vije6mppdkx6jjf&rid=giphy.gif"
  />
  <p>
    One punch
  </p>
</div>
`;
